warehouse/query-core/src/main/java/datawave/query/postprocessing/tf/TermFrequencyHitFunction.java

```diff
@@ -0,0 +1,689 @@
+        Multimap<String,Function> allFunctions = TermOffsetPopulator.getContentFunctions(script);
+        Set<Function> functions = new HashSet<>(allFunctions.values());
+        for (Function function : functions) {
+            List<SubQuery> subQueries = parseFunction(function);
+            for (SubQuery subQuery : subQueries) {
+                
+                // populate field search space
+                functionFields.add(subQuery.field);
+                
+                if (subQuery.isNegated) {
+                    negativeSubQueries.add(subQuery);
+                    for (String fieldValue : subQuery.fieldValues) {
+                        fvToFunctionIds.put(fieldValue, subQuery.id);
+                        negativeSearchSpace.add(fieldValue);
+                    }
+                } else {
+                    positiveSubQueries.add(subQuery);
+                    for (String fieldValue : subQuery.fieldValues) {
+                        fvToFunctionIds.put(fieldValue, subQuery.id);
+                        positiveSearchSpace.add(fieldValue);
+                    }
+                }
+            }
+        }
+    }
+    
+    /**
+     * Get this function's search space as defined by fields and values
+     *
+     * @param function
+     *            a {@link Function} that is either adjacent, phrase, or within
+     * @return a list of fields and values associated with the provided function
+     */
+    private List<SubQuery> parseFunction(Function function) {
+        // functions may take different forms..
+        // within = {field, number, termOffsetMap, terms...}
+        // within = {number, termOffsetMap, terms...}
+        // adjacent = {field, termOffsetMap, terms...}
+        // adjacent = {termOffsetMap, terms...}
+        // phrase = {field, termOffsetMap, terms...}
+        // phrase = {termOffsetMap, terms...}
+        List<JexlNode> args = function.args();
+        int index = function.name().equals("within") ? 3 : 2;
+        
+        // If the first arg is a number or termOffsetMap then no field was provided as part of the function.
+        // The fields will be build via value lookups from the tfFVs.
+        boolean specialCase = false;
+        JexlNode first = args.get(0);
+        if (isFirstNodeSpecial(first)) {
+            specialCase = true;
+            index--;
+        }
+        
+        // Parse the values first. Might have to lookup fields by value.
+        TreeSet<String> values = getValuesFromArgs(args, index);
+        
+        TreeSet<String> fields;
+        if (specialCase) {
+            // field(s) were not present in the function, lookup fields by value.
+            fields = lookupFieldsByValues(values);
+        } else {
+            // function's first arg is the field, or fields in the form (FIELD_A || FIELD_B)
+            fields = parseField(args.get(0));
+        }
+        
+        // delayed = negated
+        boolean isNegated = isDelayed(function);
```

Comment:
It may be more precise to say that delayed == not part of the nested iterator bridge construct. The IteratorBuildingVisitor stops at delayed markers, so if this function is under a delayed marker it's terms have not been fetched as part of the field index lookup -- thus requiring a fetch here.

Reply:
please update the comments variables to reflect this then

Reply:
Updated in next commit.